os/arch/arm/src/armv7-a/arm_assert.c

```diff
@@ -497,56 +487,37 @@ static inline void print_assert_detail(const uint8_t *filename, int lineno, stru
 	}
 #endif
 #endif /* defined(CONFIG_DEBUG_WORKQUEUE) */
-
-	lldbg_noarg("=====================================================================\n");
-	lldbg_noarg("Checking kernel heap for corruption...\n");
-	lldbg_noarg("=====================================================================\n");
-	if (mm_check_heap_corruption(g_kmmheap) == OK) {
-		lldbg_noarg("No kernel heap corruption detected\n");
-	} else {
-		lldbg_noarg("##########################################################################################################################################\n");
-		lldbg_noarg("\n\n");
-		/* treat kernel fault */
-		arm_assert();
-	}
-	lldbg_noarg("=====================================================================\n");
-	lldbg_noarg("\n");
+	up_dumpstate(fault_tcb, asserted_location);
 	
 	/* Dump the state of all tasks (if available) */
 	task_show_alivetask_list();
-	lldbg_noarg("\n");
-
-	/* Dump the asserted TCB */
-	lldbg_noarg("=====================================================================\n");
-	lldbg_noarg("Asserted TCB Info\n");
-	lldbg_noarg("=====================================================================\n");
-	task_show_tcbinfo(fault_tcb);
-	lldbg_noarg("=====================================================================\n");
-	lldbg_noarg("\n");
 
-#ifdef CONFIG_APP_BINARY_SEPARATION
-	lldbg_noarg("=====================================================================\n");
-	lldbg_noarg("Application Info\n");
-	lldbg_noarg("=====================================================================\n");
-	elf_show_all_bin_section_addr();
-	lldbg_noarg("=====================================================================\n");
-	lldbg_noarg("\n");
-#endif
 
-	up_dumpstate(fault_tcb, asserted_location);
 
 #ifdef CONFIG_APP_BINARY_SEPARATION
+	elf_show_all_bin_section_addr();
 	if (IS_FAULT_IN_USER_THREAD(fault_tcb)) {
-		lldbg_noarg("=====================================================================\n");
-		lldbg_noarg("Checking current app heap for corruption...\n");
-		lldbg_noarg("=====================================================================\n");
+		lldbg_noarg("===========================================================App_Heap_Corruption==========================================\n");
+		lldbg("Checking current app heap for corruption...\n");
 		if (mm_check_heap_corruption((struct mm_heap_s *)(fault_tcb->uheap)) == OK) {
-			lldbg_noarg("No app heap corruption detected\n");
+			lldbg("No app heap corruption detected\n");
 		}
-		lldbg_noarg("=====================================================================\n");
 	}
 #endif
 
+	/* Dump the asserted TCB */
+	task_show_tcbinfo(fault_tcb);
+
+	lldbg_noarg("===========================================================Kernel_Heap_Corruption==========================================\n");
+	lldbg("Checking kernel heap for corruption...\n");
```

Comment:
If the kernel heap check is only in print_assert_detail(), the kernel heap check is not performed when the OS is a secure state.

How about changing as below for both armv8-a and armv7-a?
```
 void print_assert_detail()
{
	
	...
	if (mm_check_heap_corruption(g_kmmheap) == OK) {
		lldbg("No kernel heap corruption detected\n");
	} else {
		/* treat kernel fault */
		arm_assert();
	}
}

...
void up_assert() {
	...

	if (!IS_SECURE_STATE()) {
		print_assert_detail(filename, lineno, fault_tcb, asserted_location);
	} else {
		if (mm_check_heap_corruption(g_kmmheap) == OK) {
			lldbg("No kernel heap corruption detected\n");
		} else {
			/* treat kernel fault */
			arm_assert();
		}
	}
```

Reply:
So, for armv7-a, the changes you suggested are already there.
but for armv8-m, the change you mentioned has kernel heap check when the OS is in secured state.
Kernel heap corruption check should be done when OS is in secure state?

Reply:
if kernel heap is corrupted, we must reboot
because entering user app recovery is meaningless

Reply:
The approach you suggested would have redundant code for heap corruption check. So, can't we do heap corruption check in v7-a as we have done in v8-m?
` void print_assert_detail()
{
	...
       // No kernel heap corruption check
}

...
void up_assert() {
	...
        if (!IS_SECURE_STATE()) {
		print_assert_detail(filename, lineno, fault_tcb, asserted_location);
	} 
	
	lldbg_noarg("===========================================================Kernel_Heap_Corruption==========================================\n");
	lldbg("Checking kernel heap for corruption...\n");
	if (mm_check_heap_corruption(g_kmmheap) == OK) {
		lldbg("No kernel heap corruption detected\n");
	} else {
		/* treat kernel fault */
		arm_assert();
	}
}`

Reply:
The approach you suggested would have redundant code for heap corruption check.
So, how about changing heap corruption check in v7-a as same as v8-m?
```
void print_assert_detail()
{
    ...
    // No kernel heap corruption check
}

...

void up_assert() {
    ...
    	if (!IS_SECURE_STATE()) {
		print_assert_detail(filename, lineno, fault_tcb, asserted_location);
	}
	lldbg_noarg("===========================================================Kernel_Heap_Corruption==========================================\n");
	lldbg("Checking kernel heap for corruption...\n");
	if (mm_check_heap_corruption(g_kmmheap) == OK) {
		lldbg("No kernel heap corruption detected\n");
	} else {
		/* treat kernel fault */
		arm_assert();
	}
}
```

Reply:
Thank you for your opinion, Looks good to me.

but when os is secure state, we must do not print below logs.
`lldbg("Checking kernel heap for corruption...\n");`
 `lldbg("No kernel heap corruption detected\n");`

Please check again.

Reply:
We have updated the code considering the fact that we don't print any log when os is secure state.